### PR TITLE
Spinlock cleanup

### PR DESCRIPTION
**Pull request by ryao**
I found a few minor improvements in a review of the use of spinlocks in the zfsonlinux/zfs repository yesterday evening.

### CODE REVIEW

**Comment by ryao** [comment]
@behlendorf Just a FYI, ryao/zfs@7046959c570b64346c00e620406a08142a74424f is safe to merge. It deletes dead code.

**Comment by ryao** [comment]
@behlendorf Good question. My guess is that we take the following branch in this situation:

http://lxr.free-electrons.com/source/kernel/sched/core.c#L7400

The control flow there is somewhat hard to follow because of how `preempt_count_equals` is written. We could try stepping through this in a VM in kgdb to see what happens.

**Comment by behlendorf** [comment]
Commit  3504ceb skipped, the other were merged as:

919efe9 zfs_inode_update should not call dmu_object_size_from_db under spinlock
bc8ffb2 Remove obsolete zv_lock
